// @shopify/cli/dist/local-IRPCVBNP.js
import {
  alwaysLogAnalytics,
  alwaysLogMetrics,
  analyticsDisabled,
  ciPlatform,
  cloudEnvironment,
  codespacePortForwardingDomain,
  codespaceURL,
  firstPartyDev,
  gitpodURL,
  hasGit,
  homeDirectory,
  isCloudEnvironment,
  isDevelopment,
  isShopify,
  isTerminalInteractive,
  isUnitTest,
  isVerbose,
  macAddress,
  opentelemetryDomain,
  useDeviceAuth,
  useEmbeddedThemeCLI,
  useThemebundling
} from "./chunk-YK3ZIR2O.js";
import "./chunk-QETDXAFX.js";
import "./chunk-F6BSBHJJ.js";
import "./chunk-6DU4BL5F.js";
import "./chunk-SNOECVP4.js";
import "./chunk-2AW6NZQ5.js";
import "./chunk-6XKWVDB6.js";
import "./chunk-UBB7JKND.js";
import "./chunk-T2GATXSP.js";
import "./chunk-XVNW332R.js";
import "./chunk-LAPJDMM4.js";
import "./chunk-7Q3MMWAC.js";
import "./chunk-F27QYMO7.js";
import "./chunk-ZFVK6EQX.js";
import "./chunk-522OB3EU.js";
import "./chunk-PNN7RS7Y.js";
import "./chunk-POZ5MGPT.js";
export {
  alwaysLogAnalytics,
  alwaysLogMetrics,
  analyticsDisabled,
  ciPlatform,
  cloudEnvironment,
  codespacePortForwardingDomain,
  codespaceURL,
  firstPartyDev,
  gitpodURL,
  hasGit,
  homeDirectory,
  isCloudEnvironment,
  isDevelopment,
  isShopify,
  isTerminalInteractive,
  isUnitTest,
  isVerbose,
  macAddress,
  opentelemetryDomain,
  useDeviceAuth,
  useEmbeddedThemeCLI,
  useThemebundling
};
